lms/models/json_settings.py

```diff
@@ -1,42 +1,59 @@
 import base64
+from collections.abc import Mapping
 from dataclasses import dataclass
+from enum import Enum, member
 from typing import Any
 
 import sqlalchemy as sa
+from pyramid.settings import asbool
 from sqlalchemy.ext.mutable import MutableDict
 from sqlalchemy.orm import InstrumentedAttribute
 
 
-@dataclass
+class SettingFormat(Enum):
+    """Valid values for JSONSetting.format.
+
+    Not that some of this value double an an identifier and a function as `f(str) -> Any` to convert strings to the right JSON value (eg. asbool)
+    while other are just mere identifiers (AS_SECRET).
+    """
+
+    BOOLEAN = member(asbool)
+
+    AES_SECRET = object()  # Helper to declare settings as secret.
```

Comment:
Note that this is not a function.

This will cause needed a type ignore further below.

I reckon the right solution is to make this all objects with explicit methods, eg. 

```
class AESFormatter:
  def AESFormater format(settings, str):
     settings.set_secret(str)
```

I rather tackling that in a separate PR, ideally once every setting is accessed via the new "system".